src/parser/jobs/rdm/modules/Gauge.tsx

```diff
@@ -0,0 +1,307 @@
+		[this.data.actions.VERFIRE.id, {white: 0, black: 9}],
+		[this.data.actions.VERFLARE.id, {white: 0, black: 21}],
+		[this.data.actions.JOLT.id, {white: 3, black: 3}],
+		[this.data.actions.JOLT_II.id, {white: 3, black: 3}],
+		[this.data.actions.IMPACT.id, {white: 3, black: 3}],
+		[this.data.actions.SCORCH.id, {white: 7, black: 7}],
+	])
+	public _spenderModifiers = new Map<number, GaugeModifier>([
+		[this.data.actions.ENCHANTED_REPRISE.id, {white: -5, black: -5}],
+		[this.data.actions.ENCHANTED_MOULINET.id, {white: -20, black: -20}],
+		[this.data.actions.ENCHANTED_RIPOSTE.id, {white: -30, black: -30}],
+		[this.data.actions.ENCHANTED_ZWERCHHAU.id, {white: -25, black: -25}],
+		[this.data.actions.ENCHANTED_REDOUBLEMENT.id, {white: -25, black: -25}],
+	])
+	private _severityWastedMana = {
+		1: SEVERITY.MINOR,
+		20: SEVERITY.MEDIUM,
+		80: SEVERITY.MAJOR,
+	}
+	private _severityLostMana = {
+		1: SEVERITY.MINOR,
+		20: SEVERITY.MEDIUM,
+		80: SEVERITY.MAJOR,
+	}
+	public readonly _manaDifferenceThreshold = 30
+	private readonly _manaLostDivisor = 2
+	private readonly _manaficationMultiplier = 2
+	public readonly _manaCap = 100
+
+	_manaStatistics = {
+		white: {
+			manaficationLoss: 0,
+			imbalanceLoss: 0,
+			invulnLoss: 0,
+		},
+		black: {
+			manaficationLoss: 0,
+			imbalanceLoss: 0,
+			invulnLoss: 0,
+		},
+	}
+
+	override initialise() {
+		super.initialise()
+
+		this.addEventHook(
+			filter<Event>()
+				.type('damage')
+				.source(this.parser.actor.id),
+			this.onGaugeModifying
+		)
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(oneOf(Array.from(this._spenderModifiers.keys()))),
+			this.onGaugeSpender
+		)
+		this.addEventHook(
+			filter<Event>()
+				.type('action')
+				.source(this.parser.actor.id)
+				.action(this.data.actions.MANAFICATION.id),
+			this.onManafication
+		)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onGaugeModifying(event: Events['damage']) {
+		if (event.cause.type !== 'action') {
+			return
+		}
+		const modifier = this._gaugeModifiers.get(event.cause.action)
+
+		if (modifier == null) {
+			return
+		}
+
+		const amount = modifier
+		const penalized = this.isOutOfBalance()
+		const whiteModified = penalized.white ? Math.ceil(amount.white/this._manaLostDivisor) : amount.white
+		const blackModified = penalized.black ? Math.ceil(amount.black/this._manaLostDivisor) : amount.black
+
+		if (!isSuccessfulHit(event)) {
+			//Then we lost this mana, add to statistics and move along.
+			this._manaStatistics.white.invulnLoss = whiteModified
+			this._manaStatistics.black.invulnLoss = blackModified
+			return
+		}
+
+		this._whiteManaGauge.modify(whiteModified)
+		this._blackManaGauge.modify(blackModified)
+		//Statistics Gathering
+		if (penalized.white) {
+			this._manaStatistics.white.imbalanceLoss = whiteModified
+		}
+		if (penalized.black) {
+			this._manaStatistics.black.imbalanceLoss = blackModified
+		}
```

Comment:
This is not adding the correct amount of loss due to imbalance, you need to look at the difference between base amount and the amount actually gained.  Also, you really could get rid of the penalized checks here, since amount.white - whiteModified will be 0 when in balance.
```suggestion
		this._manaStatistics.white.imbalanceLoss = amount.white - whiteModified
		this._manaStatistics.black.imbalanceLoss = amount.black - blackModified
```

Reply:
updated slightly differently than suggested, it wasn't doing += so it was only capturing the last penalty of the set.  Fixed the other manaStatistic because it had the same problem.